src/Sarif/ExternalProcess.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public class ExternalProcess
+    {
+        public IConsoleCapture StdOut { get; private set; }
+
+        public IConsoleCapture StdErr { get; private set; }
+
+        public ExternalProcess(
+            string workingDirectory,
+            string fileName,
+            string arguments,
+            IConsoleCapture stdOut = null,
+            int[] acceptableReturnCodes = null)
+        {
+            acceptableReturnCodes = acceptableReturnCodes ?? new int[] { 0 };
+
+            ProcessStartInfo psi = new ProcessStartInfo();
+
+            psi.FileName = fileName;
+            psi.Arguments = arguments;
+            psi.WorkingDirectory = workingDirectory;
+
+            psi.ErrorDialog = false;
+            psi.CreateNoWindow = true;
+            psi.UseShellExecute = false;
+            psi.RedirectStandardInput = true;
+            psi.RedirectStandardError = true;
+            psi.RedirectStandardOutput = true;
+            psi.StandardOutputEncoding = System.Text.Encoding.UTF8;
+            psi.WindowStyle = ProcessWindowStyle.Hidden;
+
+            using (Process process = Process.Start(psi))
+            using (ManualResetEvent mreOut = new ManualResetEvent(false), mreErr = new ManualResetEvent(false))
```

Comment:
>mreErr [](start = 74, length = 6)

LGTM complains that these two assignments are useless because neither `mreOut` nor `mreErr` is used. Is there a reason to created these events?

Reply:
This is just dead code from an earlier draft, you can remove it.

---
In reply to: [472434440](https://github.com/microsoft/sarif-sdk/pull/2035#discussion_r472434440) [](ancestors = 472434440)